### PR TITLE
Download zip of files not in same document

### PR DESCRIPTION
**Pull request by archiloque**
This is a draft for https://github.com/sismics/docs/issues/570 , can you tell me if you have feedbacks before I finish it?

### CODE REVIEW

**Comment by jendib** [comment]
I will do a full review when it's done, but it looks fine for now 👍

**Comment by archiloque** [comment]
@jendib thanks, I'll add automated tests and make it ready for review when it's done

**Comment by archiloque** [comment]
> @archiloque Would you have time to finish this please?

Sorry for the delay, I started to work on upstreaming our changes, the first is ready at https://github.com/sismics/docs/pull/582 (I'll deal with one PR at the time to make things easier to manage, but will be sure to answer quickly)

**Comment by archiloque** [comment]
Sorry it was a bad click : it's not ready for review, and I can't find the option to make it a work in progress again

File: docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java [hunk 1]
```diff
@@ -738,26 +773,60 @@ public Response zip(
...
+     * @param shareId Share ID
+     * @param file
+     */
+    private void checkFileAccessible(String shareId, File file, boolean jsonResponse) {
         if (file.getDocumentId() == null) {
             // It's an orphan file
             if (!file.getUserId().equals(principal.getId())) {
                 // But not ours
-                throw new ForbiddenClientException();
+                if (jsonResponse) {
```
Comment (jendib) [inline docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java:812]: I'm not fan of this, is there a way to always throw the same exception?

File: docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java [hunk 2]
```diff
@@ -692,12 +693,46 @@ public Response zip(
...
+            @FormParam("files") List<String> filesIdsList) {
+        authenticate();
+        List<File> fileList = findFiles(filesIdsList);
+        return sentZippedFiles("files", fileList);
+    }
+
+    /**
+     * Sent the content of a list of files.
+     */
+    private Response sentZippedFiles(String zipFileName, List<File> fileList) {
```
Comment (jendib) [inline docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java:733]: Should be "sendZippedFiles"?

File: docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java (hunk 1 above)
Comment (archiloque) [reply docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java]: I wasn't too happy about it either, after enough digging in the doc I found you could use a `MessageBodyWriter` to fix this case, which means the solution is even generic. Whet do you think about it?

File: docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java (hunk 2 above)
Comment (archiloque) [reply docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java]: You're right, fixed!

**Pull request merged**